File: 프로그래머스/코딩테스트 고득점 Kit/완전탐색/최소직사각형.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int w=0,h=0;
    for(int i = 0; i<sizes.size(); i++){
        int large = max(sizes[i][0], sizes[i][1]); //현재 명함의 가장 긴 면
        int small = min(sizes[i][0], sizes[i][1]); //현재 명함의 가장 작은 면
        w = max(w, large); //각각의 명함 중에서 가장 긴 면
        h = max(h, small); //각각의 명함 중에서 가장 짧은 면
    }

    return w*h;
}

int main(){
    vector<vector<int>> sizes = {{60, 50}, {30, 70}, {60, 30}, {80, 40}};
    cout << solution(sizes);
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;
class Solution {
    int[][] sizes;
    public int solution(int[][] sizes) {
        this.sizes = sizes;
        int answer = 0;
        
        int l1 = 0, l2 = 0;
        
        for(int[] size : sizes){
            if(size[1] > size[0]){
                int tmp = size[0];
                size[0] = size[1];
                size[1] = tmp;
            }
        }
        Arrays.sort(sizes, (o1, o2)->(o2[0]-o1[0]));
        for(int[] size : sizes){
            l1 = Math.max(l1, size[0]);
            l2 = Math.max(l2, size[1]);
        }
        answer = l1*l2;
        return answer;
    }
}


//정렬하고 가장 긴 변을 돌릴 수 있을지 확인